forge-gui/src/main/java/forge/itemmanager/advancedsearchparsers/RarityParser.java

```diff
@@ -0,0 +1,226 @@
+package forge.itemmanager.advancedsearchparsers;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Predicate;
+
+import forge.card.CardRarity;
+import forge.item.PaperCard;
+import forge.item.PaperCardPredicates;
+import forge.util.ComparableOp;
+
+public abstract class RarityParser {
+    private static final Map<CardRarity, Integer> RARITY_RANK;
+    static {
+        RARITY_RANK = new HashMap<>();
+        RARITY_RANK.put(CardRarity.BasicLand, 0);
+        RARITY_RANK.put(CardRarity.Common, 0);
+        RARITY_RANK.put(CardRarity.Uncommon, 1);
+        RARITY_RANK.put(CardRarity.Rare, 2);
+        RARITY_RANK.put(CardRarity.MythicRare, 3);
+    }
+
+    /**
+     * Handles exact rarity check of a card
+     * @param tokenValue Token value
+     * @return Predicate or null
+     */
+    public static Predicate<PaperCard> handleExact(String tokenValue) {
+        Predicate<PaperCard> predicate = null;
+        switch(tokenValue) {
+            case "c":
+            case "common":
```

Comment:
Add case for "l" "land"

also, the following methods have duplicated code,
probably extract the "string" to CardRarity into an extra function